lib/parser/attribute-parser.ts

```diff
@@ -1,19 +1,72 @@
-import { OAttribute, ObjectBase } from "./objects";
+import { OLexerToken } from "../lexer";
+import { ExpressionParser } from "./expression-parser";
+import { OAttributeDeclaration, OAttributeSpecification, ObjectBase, OReference, ParserError } from "./objects";
 import { ParserBase, ParserState } from "./parser-base";
 
 export class AttributeParser extends ParserBase {
-  constructor(state: ParserState, private parent: ObjectBase ) {
+  constructor(state: ParserState, private parent: ObjectBase) {
     super(state);
     this.debug(`start`);
 
   }
-  parse(): OAttribute {
-    this.expect('attribute');
-    const attribute = new OAttribute(this.parent, this.state.pos.getRangeToEndLine());
-    attribute.lexerToken = this.consumeToken();
-    this.advanceSemicolon(true);
+  parse() {
+    const attribute = this.expect('attribute');
+    const token = this.consumeToken();
+    if (this.getToken().getLText() === 'of') {
+      return this.parseAttributeSpecification(token, attribute);
+    } else if (this.getToken().getLText() === ':') {
+      return this.parseAttributeDeclaration(token, attribute);
+    } else {
+      throw new ParserError(`Unexpected token ${this.getToken().text} in AttributeParser`, this.getToken().range);
+    }
+  }
+  parseAttributeDeclaration(identifier: OLexerToken, attribute: OLexerToken) {
+    this.expect(':');
+    const [tokens, semicolon] = this.advanceParenthesisAware([';'], true, true);
+    if (tokens.length === 0) {
+      throw new ParserError('type_mark expected for attribute_declaration', semicolon.range);
```

Comment:
This definitely can be just an error message.
```suggestion
      this.state.messages.push({
        message: 'type_mark expected for attribute_declaration',
        range: semicolon.range,
        severity: DiagnosticSeverity.Error
      });
      return undefined;
```